core/src/main/java/com/linecorp/armeria/internal/annotation/AnnotatedHttpService.java

```diff
@@ -232,19 +235,48 @@ public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exc
+                if (useBlocking) {
+                    return f.thenApplyAsync(
+                            msg -> new ExceptionFilteredHttpResponse(ctx, req,
+                                                                     (HttpResponse) invoke(ctx, req, msg),
+                                                                     exceptionHandler),
+                            ctx.blockingTaskExecutor());
+                } else {
+                    return f.thenApply(
+                            msg -> new ExceptionFilteredHttpResponse(ctx, req,
+                                                                     (HttpResponse) invoke(ctx, req, msg),
+                                                                     exceptionHandler));
+                }
+
             case COMPLETION_STAGE:
-                return f.thenCompose(msg -> toCompletionStage(invoke(ctx, req, msg)))
-                        .handle((result, cause) -> cause == null ? convertResponse(ctx, req, null, result,
-                                                                                   HttpHeaders.of())
-                                                                 : exceptionHandler.handleException(ctx, req,
-                                                                                                    cause));
+                if (useBlocking) {
+                    return f.thenComposeAsync(msg -> toCompletionStage(invoke(ctx, req, msg)),
+                                              ctx.blockingTaskExecutor())
+                            .handle((result, cause) ->
+                                            cause == null ? convertResponse(ctx, req, null, result,
+                                                                            HttpHeaders.of())
+                                                          : exceptionHandler.handleException(ctx, req, cause));
+                } else {
+                    return f.thenCompose(msg -> toCompletionStage(invoke(ctx, req, msg)))
+                            .handle((result, cause) ->
+                                            cause == null ? convertResponse(ctx, req, null, result,
+                                                                            HttpHeaders.of())
+                                                          : exceptionHandler.handleException(ctx, req, cause));
+                }
+
             default:
-                return f.thenApplyAsync(msg -> convertResponse(ctx, req, null, invoke(ctx, req, msg),
-                                                               HttpHeaders.of()),
-                                        ctx.blockingTaskExecutor());
+                if (useBlocking) {
+                    logger.warn("The current thread is using blockTaskExecutor. It'll migrate to use" +
+                                " event loop thread when soon. Therefore, use @Blocking to avoid" +
+                                " unexpected problems.");
```

Comment:
How about adding the name so that a user easily notice? `object.getClass().getSimpleName() + '#' + method.getName()`

Reply:
Fixed 9cdc252